### DISCUSSION TITLE
How to setup relationship with auth.users and other public tables using drizzle

### DISCUSSION THREAD

**Discussion by mkappworks** [supabase/supabase#27426] [Questions] [answered]
I am working on a nextjs app with supabase auth, db and drizzle ORM.
I am looking for a method to define relationship between auth.users and other public tables using drizzle.

Appreciate any answers

**Comment by GaryAustin1** [comment]
Not sure about Drizzle.  But one approach I've seen done for Prisma is to use the user management auth.users trigger guide to set up a user table in a public or your own schema that you control.  Then you use that table for you schema and info you need.

**Reply by mkappworks** [reply]
I think you are referring to this guide https://supabase.com/docs/guides/auth/managing-user-data

1. Can we write the triggers using javascript or this should be done via the supabase dashboard?
2. Is there a way to define this sql in drizzle. I think the real question is how to get a reference of auth.users in drizzle schema?

```
create table public.profiles (
  id uuid not null references auth.users on delete cascade,
  first_name text,
  last_name text,

  primary key (id)
);

alter table public.profiles enable row level security;
```

**Reply by GaryAustin1** [reply, marked as answer]
I think the approach would be to make that profile table (or user table) in your Drizzle code.  The fk is not required if that causes issues for Drizzle (as it is to a non Drizzle owned schema).  Note I know nothing about Drizzle so mainly thinking about what Prisma users do which I assume is similar.  

Then you use an auth.users trigger and function done with the SB dashboard or any SQL approach you desire.  Basically you use SQL to add a trigger and call a function.  That function inserts data to your Drizzle table whenever Supabase auth inserts a new user into there table.

Then you have relation links from you other tables to this new user table.
